### PR TITLE
Namespace metrics under "app"

### PR DESCRIPTION
**Pull request by jason-o-matic**
Metrics as they existed before conflicted in some ways with
InstrumentalD. This clarifies that these metrics come from your
application code.

This also clarifies the service and exception namespaces, making them
more specific.

### CODE REVIEW

File: METRICS.MD [hunk 1]
```diff
@@ -5,21 +5,21 @@
 * `active_record.users.find`
```
Comment (janxious) [inline METRICS.MD]: There is a bunch of outdated information in here, and I think when the README links to the production metrician documentation, this file is probably redundant and can be deleted entirely. Thoughts?

**Comment by janxious** [comment]
It occurred to me in a sleep vision that we could change none of the code in the gem and have a default prefix of "app".

`Metrician.prefix = "app"`

I think that works better than this solution in some ways, since in the world of this PR, if someone sets a prefix, they are going to get "<prefix>.app.", with no way to remove that leader.

**Comment by janxious** [comment]
Also that could be set in the agent configuration instead of directly in metrician, which could be interesting.

**Comment by jason-o-matic** [comment]
Updated if you want to gander again @janxious!

File: lib/metrician.rb [hunk 2]
```diff
@@ -53,12 +54,13 @@ def self.dotify(klass)
...
   # TODO: consider removal/movement to Instrumental Agent
   def self.prefix=(prefix)
+    @prefixed = nil
     @prefix = prefix.to_s[-1] == "." ? prefix.to_s : "#{prefix}."
   end
 
   # TODO: consider removal/movement to Instrumental Agent
   def self.prefix
-    @prefix || ""
+    @prefix || DEFAULT_PREFIX
```
Comment (janxious) [inline lib/metrician.rb]: Maybe this should be `||=` ?

File: lib/metrician.rb [hunk 3]
```diff
@@ -53,12 +54,13 @@ def self.dotify(klass)
 
   # TODO: consider removal/movement to Instrumental Agent
```
Comment (janxious) [inline lib/metrician.rb]: I think all the TODO comments around prefixes can be removed

**Review by janxious — APPROVED** [review]
I think there are some small changes to be made, but this is looking right.

**Pull request merged**